src/pages/index.js

```diff
@@ -36,26 +133,110 @@ function handleImageClick(name, link) {
   imagePopup.open({ name, link });
 }
 
-// New PopupWithForm instances (userinfo integration change)
+// New PopupWithForm instances (userinfo integration change step 3)
 const editProfilePopup = new PopupWithForm("#profile-edit-modal", (data) => {
-  userInfo.setUserInfo({ name: data.title, job: data.description });
-  editProfilePopup.close(); //closes popup after editing profile
+  const submitButton = editProfilePopup._form.querySelector(".modal__button");
+  const originalButtonText = submitButton.textContent;
+  submitButton.textContent = "Saving...";
+
+  return api
+    .setUserInfo({
+      name: data.title,
+      about: data.description,
+    })
+    .then((updatedUserData) => {
+      userInfo.setUserInfo({
+        name: updatedUserData.name,
+        job: updatedUserData.about,
+      });
+      return new Promise((resolve) => setTimeout(resolve, 1000)); // 1s delay
+    })
+    .then(() => {
+      editProfilePopup.close();
```

Comment:
`      editProfilePopup.close();
`

It's not needed here because you have ` this.close();` inside `PopupWithForm`